### PR TITLE
quantization: fix scale+zp serialization of quantized BatchNorm{2|3}d

### LINKED ISSUE #43774: investigate reported issues with quantized BatchNorm2d
## 🐛 Bug

In https://discuss.pytorch.org/t/incorrect-results-after-loading-saved-quantized-model/93844, there is a report of incorrect behavior of unfused quantized BatchNorm2d.  We should verify that it works correctly.

## To Reproduce

1.
1.
1.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70432

Summary:

Scale and zero_point need to be buffers for serialization to work
on them properly.  This PR moves them to buffers.  This is BC breaking,
but the "before" state was completely broken (scale + zp were not
serialized at all) so there is no value in trying to handle it.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm3d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33330022](https://our.internmc.facebook.com/intern/diff/D33330022)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

File: test/quantization/core/test_quantized_module.py [hunk 1]
```diff
@@ -644,6 +644,50 @@ def test_batch_norm3d(self):
         self.assertEqual(quant_ref.int_repr().numpy(), qy.int_repr().numpy(),
                          msg="BatchNorm3d module API failed")
 
+    def _test_batch_norm_serialization(self, get_model, data1, data2):
+        m1 = get_model()
+        m1.qconfig = torch.quantization.default_qconfig
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_module.py:649]: nit: torch.ao.quantization

**Review by jerryzh168 — APPROVED** [review]
Looks good, thanks!

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).
